# Google Apps Analysis

<a id="table_of_contents"></a>

### Table of contents

<ol>
  <li><a href="#overview">Situation Overview</a>
  <ul>
   <li><a href='#dataset'>Examine the overall dataset
    </a></li>
    <li><a href='#cleaning'>Execute cleaning procedures
    </a></li>
    
  </ul>
  </li>
  
  <li><a href="#eda">Exploratory analysis</a>
  <ul>
    <li><a href='#unique_values'>Begin by computing distinct values
    </a></li>
    <li><a href='#null_values'>Identify any null values within the dataset
    </a></li>
    <li><a href='#apps_by_category'>Find out the number of apps by category
    </a></li>
    <li><a href='#apps_by_rating'>Find out the app ratings: minimum, maximum, and average
    </a></li>
    </ul>
  </li>  
  
  <li><a href="#analysis">Data analysis</a>
  <ul>
    <li><a href='#free_paid_apps'>Comparing ratings, reviews and installs between free and paid apps
    </a></li>
    <li><a href='#low_rating_apps'>Examine apps with low ratings and reviews by categories
    </a></li>
    <li><a href='#negative_sentiment'>Examine apps with highest negative sentiment grouped by app name and app category
    </a></li>
    <li><a href='#high_rating_apps'>Examine highly rated apps by categories
    </a></li>
    <li><a href='#high_installed_apps'>Examine highly installed apps by categories
    </a></li>
    <li><a href='#leading_apps_by_installs'>Evaluate top apps in each category based on installation numbers
    </a></li>
    <li><a href='#leading_apps'>Evaluate leading apps by category: price, user ratings, average reviews
    </a></li>
    </ul>
  </li>  
  <li><a href="#insights">Insights</a>
</ol>



<a id="overview"></a>
## Situation Overview
An application developer objective is to create a successful Andrioid App, in order to make an informed decision the developer wants to answer the following questions:

1. What app categories are the most popular?
2. Which is the most optimal price?
3. Which is the best way to maximize users ratings?

<a href="#table_of_contents">Navigate to contents</a>


<a id="dataset"></a>
Examine the overall dataset
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [4]:
#load sql extension
%load_ext sql

In [5]:
#connect to mysql database
%sql mysql://root:NilArj_21@localhost:3306/project

In [12]:
%%sql 
select * from googleplaystore 
limit 3

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', 4.1, '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'),
 ('Coloring book moana', 'ART_AND_DESIGN', 3.9, '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up'),
 ('U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', 4.7, '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up')]

In [13]:
%%sql 
select * from googleplaystore_user_reviews
limit 3

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('10 Best Foods for You', 'I like eat delicious food. That\'s I\'m cooking food myself, case "10 Best Foods" helps lot, also "Best Before (Shelf Life)"', 'Positive', 1.0, 0.5333333333333333),
 ('10 Best Foods for You', 'This help eating healthy exercise regular basis', 'Positive', 0.25, 0.2884615384615384),
 ('10 Best Foods for You', None, None, None, None)]

In [14]:
%%sql
select column_name, data_type
from information_schema.columns
where table_schema = "project"
    and table_name = "googleplaystore"

 * mysql://root:***@localhost:3306/project
13 rows affected.


[('Android Ver', 'text'),
 ('App', 'text'),
 ('Category', 'text'),
 ('Content Rating', 'text'),
 ('Current Ver', 'text'),
 ('Genres', 'text'),
 ('Installs', 'text'),
 ('Last Updated', 'text'),
 ('Price', 'text'),
 ('Rating', 'double'),
 ('Reviews', 'text'),
 ('Size', 'text'),
 ('Type', 'text')]

In [15]:
%%sql
select column_name, data_type
from information_schema.columns
where table_schema = "project"
    and table_name = "googleplaystore_user_reviews"

 * mysql://root:***@localhost:3306/project
5 rows affected.


[('App', 'text'),
 ('Sentiment', 'text'),
 ('Sentiment_Polarity', 'double'),
 ('Sentiment_Subjectivity', 'double'),
 ('Translated_Review', 'text')]

<a id="cleaning"></a>
Execute cleaning procedures
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [16]:
%%sql 
alter table googleplaystore
modify column Reviews float

 * mysql://root:***@localhost:3306/project
(MySQLdb.DataError) (1265, "Data truncated for column 'Reviews' at row 10473")
[SQL: alter table googleplaystore
modify column Reviews float]
(Background on this error at: https://sqlalche.me/e/20/9h9h)


In [17]:
%%sql 
select * 
from googleplaystore
where Reviews is not null and Reviews regexp "[a-zA-Z]+"

 * mysql://root:***@localhost:3306/project
1 rows affected.


[('Life Made WI-Fi Touchscreen Photo Frame', '1.9', 19.0, '3.0M', '1,000+', 'Free', '0', 'Everyone', None, 'February 11, 2018', '1.0.19', '4.0 and up', None)]

In [38]:
%%sql
update googleplaystore
set Category = null, Rating = 1.9, Reviews = 19, Size = "3.0M", Installs = "1,000+", Type = "Free", Price = "0", `Content Rating` ="Everyone", Genres = null, `Last Updated` = "February 11, 2018", `Current Ver` = "1.0.19", `Android Ver`="4.0 and up"	 
where App = "Life Made WI-Fi Touchscreen Photo Frame"


 * mysql://root:***@localhost:3306/project
1 rows affected.


[]

In [39]:
%%sql 
alter table googleplaystore
modify column Reviews int

 * mysql://root:***@localhost:3306/project
10841 rows affected.


[]

In [24]:
%%sql
alter table googleplaystore
add column updated_price float

 * mysql://root:***@localhost:3306/project
0 rows affected.


[]

In [25]:
%%sql
update googleplaystore
set updated_price = case
                        when Price regexp "^$" then replace(Price, "$","")
                        else cast(Price as float)
                    end

 * mysql://root:***@localhost:3306/project
10841 rows affected.


[]

In [99]:
%%sql
alter table googleplaystore
add column updated_installs int

 * mysql://root:***@localhost:3306/project
0 rows affected.


[]

In [109]:
%%sql
update googleplaystore
set updated_installs = regexp_replace(Installs,"[+,]","")

 * mysql://root:***@localhost:3306/project
10841 rows affected.


[]

In [110]:
%%sql
select column_name, data_type
from information_schema.columns
where table_schema = "project"
    and table_name = "googleplaystore"

 * mysql://root:***@localhost:3306/project
15 rows affected.


[('Android Ver', 'text'),
 ('App', 'text'),
 ('Category', 'text'),
 ('Content Rating', 'text'),
 ('Current Ver', 'text'),
 ('Genres', 'text'),
 ('Installs', 'text'),
 ('Last Updated', 'text'),
 ('Price', 'text'),
 ('Rating', 'double'),
 ('Reviews', 'int'),
 ('Size', 'text'),
 ('Type', 'text'),
 ('updated_installs', 'int'),
 ('updated_price', 'float')]

In [111]:
%%sql 
select * from googleplaystore 
limit 3

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', 4.1, 159, '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up', 0.0, 10000),
 ('Coloring book moana', 'ART_AND_DESIGN', 3.9, 967, '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up', 0.0, 500000),
 ('U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', 4.7, 87510, '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up', 0.0, 5000000)]

## Exploratory analysis

<a id="unique_values"></a>
Begin by computing distinct values
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [28]:
%%sql
select count(distinct App) as uniqueApps
from googleplaystore

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(9638,)]

In [66]:
%%sql
select Installs,count(distinct Installs)
from googleplaystore
group by Installs

 * mysql://root:***@localhost:3306/project
21 rows affected.


[('0', 1),
 ('0+', 1),
 ('1,000,000,000+', 1),
 ('1,000,000+', 1),
 ('1,000+', 1),
 ('1+', 1),
 ('10,000,000+', 1),
 ('10,000+', 1),
 ('10+', 1),
 ('100,000,000+', 1),
 ('100,000+', 1),
 ('100+', 1),
 ('5,000,000+', 1),
 ('5,000+', 1),
 ('5+', 1),
 ('50,000,000+', 1),
 ('50,000+', 1),
 ('50+', 1),
 ('500,000,000+', 1),
 ('500,000+', 1),
 ('500+', 1)]

In [79]:
%%sql
select count(distinct Category) as uniqueCategories
from googleplaystore

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(33,)]

In [41]:
%%sql
select count(distinct App) as uniqueAppsReviews
from googleplaystore_user_reviews

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(1074,)]

In [69]:
%%sql
select Sentiment, count(distinct Sentiment) as uniqueSentiment
from googleplaystore_user_reviews
group by Sentiment

 * mysql://root:***@localhost:3306/project
4 rows affected.


[(None, 0), ('Negative', 1), ('Neutral', 1), ('Positive', 1)]

<a id="null_values"></a>
Identify any null values within the dataset
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [31]:
%%sql
select count(*) as nullValues
from googleplaystore
where App = null

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(0,)]

In [32]:
%%sql
select count(*) as nullValuesReviews
from googleplaystore_user_reviews
where App = null

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(0,)]

<a id="apps_by_category"></a>
Find out the number of apps by category
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [81]:
%%sql 
select Category, count(*) as catCount
from googleplaystore 
group by Category
order by catCount desc

 * mysql://root:***@localhost:3306/project
34 rows affected.


[('FAMILY', 1972),
 ('GAME', 1144),
 ('TOOLS', 843),
 ('MEDICAL', 463),
 ('BUSINESS', 460),
 ('PRODUCTIVITY', 424),
 ('PERSONALIZATION', 392),
 ('COMMUNICATION', 387),
 ('SPORTS', 384),
 ('LIFESTYLE', 382),
 ('FINANCE', 366),
 ('HEALTH_AND_FITNESS', 341),
 ('PHOTOGRAPHY', 335),
 ('SOCIAL', 295),
 ('NEWS_AND_MAGAZINES', 283),
 ('SHOPPING', 260),
 ('TRAVEL_AND_LOCAL', 258),
 ('DATING', 234),
 ('BOOKS_AND_REFERENCE', 231),
 ('VIDEO_PLAYERS', 175),
 ('EDUCATION', 156),
 ('ENTERTAINMENT', 149),
 ('MAPS_AND_NAVIGATION', 137),
 ('FOOD_AND_DRINK', 127),
 ('HOUSE_AND_HOME', 88),
 ('AUTO_AND_VEHICLES', 85),
 ('LIBRARIES_AND_DEMO', 85),
 ('WEATHER', 82),
 ('ART_AND_DESIGN', 65),
 ('EVENTS', 64),
 ('COMICS', 60),
 ('PARENTING', 60),
 ('BEAUTY', 53),
 (None, 1)]

<a id="apps_by_rating"></a>
Find out the app ratings: minimum, maximum, and average
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [34]:
%%sql
select min(Rating) as minRating,
       max(Rating) as maxRating,
       round(avg(Rating),2) as avgRating
from googleplaystore

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(1.0, 5.0, 4.19)]

## Data analysis

<a id="free_paid_apps"></a>
Comparing ratings, reviews and installs between free and paid apps
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [116]:
%%sql
select Type, round(avg(Rating),2) as avgRating, round(avg(Reviews),2) as avgReviews, round(avg(updated_installs)) as avgInstalls
from googleplaystore
group by Type

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('Free', 4.19, Decimal('478613.42'), Decimal('16689291')),
 ('Paid', 4.27, Decimal('11673.31'), Decimal('91195')),
 (None, None, Decimal('0.00'), Decimal('0'))]

<a id="low_rating_apps"></a>
Examine apps with low ratings and reviews by categories
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [84]:
%%sql 
select Category,round(avg(Rating),2) as avgRating, round(avg(Reviews),2) as avgReviews, round(avg(updated_price),2) as avgPrice, round(avg(updated_installs)) as avgInstalls
from googleplaystore
group by Category
order by avgRating asc, avgReviews asc
limit 6

 * mysql://root:***@localhost:3306/project
6 rows affected.


[(None, 1.9, Decimal('19.00'), 0.0, Decimal('1000')),
 ('DATING', 3.97, Decimal('31159.31'), 0.0, Decimal('1129533')),
 ('MAPS_AND_NAVIGATION', 4.05, Decimal('223790.18'), 0.0, Decimal('5286729')),
 ('TOOLS', 4.05, Decimal('324062.93'), 0.0, Decimal('13585732')),
 ('VIDEO_PLAYERS', 4.06, Decimal('630743.93'), 0.0, Decimal('35554301')),
 ('LIFESTYLE', 4.09, Decimal('33724.57'), 0.0, Decimal('1407444'))]

<a id="negative_sentiment"></a>
Examine apps with highest negative sentiment grouped by app name and app category
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [64]:
%%sql
select count(*) as reviewsCount, s.Category, r.App, round(avg(r.Sentiment_Polarity),2) as avgSentimentPolarity, avg(s.updated_price) as avgPrice
from googleplaystore_user_reviews r
left join googleplaystore s on r.App = s.App
where r.Sentiment = "Negative"
group by s.Category, r.App
order by reviewsCount desc
limit 15

 * mysql://root:***@localhost:3306/project
15 rows affected.


[(756, 'GAME', 'Candy Crush Saga', -0.14, 0.0),
 (735, 'GAME', 'Angry Birds Classic', -0.18, 0.0),
 (714, 'GAME', 'Bowmasters', -0.18, 0.0),
 (636, 'GAME', '8 Ball Pool', -0.17, 0.0),
 (348, 'GAME', 'Helix Jump', -0.09, 0.0),
 (336, 'SPORTS', 'CBS Sports App - Scores, News, Stats & Watch Live', -0.26, 0.0),
 (324, 'GAME', 'Garena Free Fire', -0.18, 0.0),
 (288, 'GAME', 'Candy Crush Soda Saga', -0.18, 0.0),
 (284, 'GAME', 'Block Puzzle', -0.15, 0.0),
 (260, 'HEALTH_AND_FITNESS', 'Calorie Counter - MyFitnessPal', -0.19, 0.0),
 (220, 'TRAVEL_AND_LOCAL', 'Expedia Hotels, Flights & Car Rental Travel Deals', -0.31, 0.0),
 (200, 'GAME', 'Bubble Shooter', -0.19, 0.0),
 (196, 'GAME', 'Flow Free', -0.14, 0.0),
 (188, 'PHOTOGRAPHY', 'Google Photos', -0.2, 0.0),
 (174, 'GAME', 'Clash Royale', -0.11, 0.0)]

<a id="high_rating_apps"></a>
Examine highly rated apps by categories
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [83]:
%%sql 
with AvgTotalRating as (
    select round(avg(Rating),2) as avgTotalRating
    from googleplaystore
)

select s.Category,
       round(avg(s.Rating),2) as avgRating, 
       round(avg(s.updated_price),2) as avgPrice,
       AvgTotalRating.avgTotalRating
from googleplaystore s
cross join AvgTotalRating
group by s.Category, AvgTotalRating.avgTotalRating
order by avgRating desc
limit 10

 * mysql://root:***@localhost:3306/project
10 rows affected.


[('EVENTS', 4.44, 0.0, 4.19),
 ('EDUCATION', 4.39, 0.0, 4.19),
 ('ART_AND_DESIGN', 4.36, 0.0, 4.19),
 ('BOOKS_AND_REFERENCE', 4.35, 0.0, 4.19),
 ('PERSONALIZATION', 4.34, 0.0, 4.19),
 ('PARENTING', 4.3, 0.0, 4.19),
 ('GAME', 4.29, 0.0, 4.19),
 ('BEAUTY', 4.28, 0.0, 4.19),
 ('HEALTH_AND_FITNESS', 4.28, 0.0, 4.19),
 ('SOCIAL', 4.26, 0.0, 4.19)]

<a id="high_installed_apps"></a>
Examine highly installed apps by categories
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [53]:
%%sql 
select Category, round(avg(updated_installs)) as avgInstalls , round(avg(updated_price),2) as avgPrice
from googleplaystore
group by Category
order by avgInstalls desc
limit 10

 * mysql://root:***@localhost:3306/project
10 rows affected.


[('COMMUNICATION', Decimal('84359887'), 0.0),
 ('SOCIAL', Decimal('47694467'), 0.0),
 ('VIDEO_PLAYERS', Decimal('35554301'), 0.0),
 ('PRODUCTIVITY', Decimal('33434178'), 0.0),
 ('GAME', Decimal('30669602'), 0.0),
 ('PHOTOGRAPHY', Decimal('30114172'), 0.0),
 ('TRAVEL_AND_LOCAL', Decimal('26623594'), 0.0),
 ('NEWS_AND_MAGAZINES', Decimal('26488755'), 0.0),
 ('ENTERTAINMENT', Decimal('19256107'), 0.0),
 ('TOOLS', Decimal('13585732'), 0.0)]

In [ ]:
%%sql 
select count(*) as reviewsCount, Installs, Category, App, Rating, Reviews, updated_price, round(avg(Sentiment_Polarity),2) as avgSentimentPolarity
from googleplaystore_user_reviews
group by App
order by reviewsCount desc, s
limit 10

<a id="leading_apps_by_installs"></a>
Evaluate top apps in each category based on installation numbers
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [157]:
%%sql 
select distinct subqueryRank, Installs, Category, App, Rating, Reviews, updated_price
from 
    (select 
        Installs,
        Category,
        App, 
        Rating, 
        Reviews, 
        updated_price,
        updated_installs,
        rank() over(partition by Category order by updated_installs desc, Rating desc, Reviews desc, App) as subqueryRank
    from googleplaystore) as subqueryRank
where subqueryRank = 1

 * mysql://root:***@localhost:3306/project
34 rows affected.


[(1, '1,000+', None, 'Life Made WI-Fi Touchscreen Photo Frame', 1.9, 19, 0.0),
 (1, '50,000,000+', 'ART_AND_DESIGN', 'Sketch - Draw & Paint', 4.5, 215644, 0.0),
 (1, '10,000,000+', 'AUTO_AND_VEHICLES', 'AutoScout24 - used car finder', 4.4, 186648, 0.0),
 (1, '10,000,000+', 'BEAUTY', 'Beauty Camera - Selfie Camera', 4.0, 113715, 0.0),
 (1, '1,000,000,000+', 'BOOKS_AND_REFERENCE', 'Google Play Books', 3.9, 1433233, 0.0),
 (1, '100,000,000+', 'BUSINESS', 'OfficeSuite : Free Office + PDF Editor', 4.3, 1002861, 0.0),
 (1, '10,000,000+', 'COMICS', 'LINE WEBTOON - Free Comics', 4.5, 1013944, 0.0),
 (1, '1,000,000,000+', 'COMMUNICATION', 'WhatsApp Messenger', 4.4, 69119312, 0.0),
 (1, '10,000,000+', 'DATING', 'Find Real Love — YouLove Premium Dating', 4.5, 212626, 0.0),
 (1, '100,000,000+', 'EDUCATION', 'Duolingo: Learn Languages Free', 4.7, 6290507, 0.0),
 (1, '1,000,000,000+', 'ENTERTAINMENT', 'Google Play Games', 4.3, 7165362, 0.0),
 (1, '5,000,000+', 'EVENTS', 'Ticketmaster Event Tickets', 4.0, 40113, 0.0),
 (1, '1,000,000,000+', 'FAMILY', 'Google Play Games', 4.3, 7168735, 0.0),
 (1, '100,000,000+', 'FINANCE', 'Google Pay', 4.2, 348132, 0.0),
 (1, '10,000,000+', 'FOOD_AND_DRINK', "Domino's Pizza USA", 4.7, 1032935, 0.0),
 (1, '1,000,000,000+', 'GAME', 'Subway Surfers', 4.5, 27725352, 0.0),
 (1, '500,000,000+', 'HEALTH_AND_FITNESS', 'Samsung Health', 4.3, 480208, 0.0),
 (1, '10,000,000+', 'HOUSE_AND_HOME', 'Houzz Interior Design Ideas', 4.6, 353813, 0.0),
 (1, '10,000,000+', 'LIBRARIES_AND_DEMO', 'MX Player Codec (ARMv7)', 4.3, 332083, 0.0),
 (1, '100,000,000+', 'LIFESTYLE', 'Tinder', 4.0, 2789775, 0.0),
 (1, '100,000,000+', 'MAPS_AND_NAVIGATION', 'Waze - GPS, Maps, Traffic Alerts & Live Navigation', 4.6, 7232629, 0.0),
 (1, '5,000,000+', 'MEDICAL', 'My Calendar - Period Tracker', 4.7, 156410, 0.0),
 (1, '1,000,000,000+', 'NEWS_AND_MAGAZINES', 'Google News', 3.9, 878065, 0.0),
 (1, '10,000,000+', 'PARENTING', 'Pregnancy Tracker & Countdown to Baby Due Date', 4.7, 658087, 0.0),
 (1, '100,000,000+', 'PERSONALIZATION', 'CM Launcher 3D - Theme, Wallpapers, Efficient', 4.6, 6702776, 0.0),
 (1, '1,000,000,000+', 'PHOTOGRAPHY', 'Google Photos', 4.5, 10859051, 0.0),
 (1, '1,000,000,000+', 'PRODUCTIVITY', 'Google Drive', 4.4, 2731211, 0.0),
 (1, '100,000,000+', 'SHOPPING', 'AliExpress - Smarter Shopping, Better Living', 4.6, 5917485, 0.0),
 (1, '1,000,000,000+', 'SOCIAL', 'Instagram', 4.5, 66577448, 0.0),
 (1, '100,000,000+', 'SPORTS', 'Dream League Soccer 2018', 4.6, 9873470, 0.0),
 (1, '1,000,000,000+', 'TOOLS', 'Google', 4.4, 8033493, 0.0),
 (1, '1,000,000,000+', 'TRAVEL_AND_LOCAL', 'Maps - Navigate & Explore', 4.3, 9235373, 0.0),
 (1, '1,000,000,000+', 'VIDEO_PLAYERS', 'YouTube', 4.3, 25655304, 0.0),
 (1, '50,000,000+', 'WEATHER', 'GO Weather - Widget, Theme, Wallpaper, Efficient', 4.5, 1422858, 0.0)]

<a id="leading_apps"></a>
Evaluate leading apps by category: price, user ratings, average reviews
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [8]:
%%sql 
select distinct subqueryRank, Installs, Category, App, Rating, Reviews, updated_price
from 
    (select 
        Installs,
        Category,
        App, 
        Rating, 
        Reviews, 
        updated_price,
        updated_installs,
        rank() over(partition by Category order by Rating desc, updated_installs desc, Reviews desc, App) as subqueryRank
    from googleplaystore) as subqueryRank
where subqueryRank = 1

 * mysql://root:***@localhost:3306/project
34 rows affected.


[(1, '1,000+', None, 'Life Made WI-Fi Touchscreen Photo Frame', 1.9, 19, 0.0),
 (1, '100+', 'ART_AND_DESIGN', 'Spring flowers theme couleurs d t space', 5.0, 1, 0.0),
 (1, '1,000,000+', 'AUTO_AND_VEHICLES', 'Tickets + PDA 2018 Exam', 4.9, 197136, 0.0),
 (1, '1,000,000+', 'BEAUTY', 'ipsy: Makeup, Beauty, and Tips', 4.9, 49790, 0.0),
 (1, '1,000+', 'BOOKS_AND_REFERENCE', 'Tozer Devotional -Series 1', 5.0, 5, 0.0),
 (1, '1,000+', 'BUSINESS', 'CK Employee Portal', 5.0, 7, 0.0),
 (1, '5,000+', 'COMICS', 'Superheroes, Marvel, DC, Comics, TV, Movies News', 5.0, 34, 0.0),
 (1, '100+', 'COMMUNICATION', 'chat dz', 5.0, 8, 0.0),
 (1, '1,000+', 'DATING', 'American Girls Mobile Numbers', 5.0, 5, 0.0),
 (1, '1,000,000+', 'EDUCATION', 'Learn Japanese, Korean, Chinese Offline & Free', 4.9, 133136, 0.0),
 (1, '1,000,000+', 'ENTERTAINMENT', '🔥 Football Wallpapers 4K | Full HD Backgrounds 😍', 4.7, 11661, 0.0),
 (1, '500+', 'EVENTS', 'SUMMER SONIC app', 5.0, 4, 0.0),
 (1, '10,000+', 'FAMILY', 'Ek Bander Ne Kholi Dukan', 5.0, 10, 0.0),
 (1, '1,000+', 'FINANCE', 'FD Calculator (EMI, SIP, RD & Loan Eligilibility)', 5.0, 104, 0.0),
 (1, '100+', 'FOOD_AND_DRINK', 'Food-Aw - Order Food Online in Aruba', 5.0, 1, 0.0),
 (1, '100+', 'GAME', "Mad Dash Fo' Cash", 5.0, 14, 0.0),
 (1, '500+', 'HEALTH_AND_FITNESS', 'Bacterial vaginosis Treatment - Sexual disease', 5.0, 2, 0.0),
 (1, '100,000+', 'HOUSE_AND_HOME', 'Viva Decora - Decoration, Photos, Architecture, House', 4.8, 3167, 0.0),
 (1, '1,000+', 'LIBRARIES_AND_DEMO', 'Eternal life', 5.0, 26, 0.0),
 (1, '5,000+', 'LIFESTYLE', 'Oración CX', 5.0, 103, 0.0),
 (1, '1,000+', 'MAPS_AND_NAVIGATION', 'BL Taxi', 4.9, 124, 0.0),
 (1, '1,000+', 'MEDICAL', 'BP Journal - Blood Pressure Diary', 5.0, 6, 0.0),
 (1, '1,000+', 'NEWS_AND_MAGAZINES', 'BI News', 5.0, 22, 0.0),
 (1, '10+', 'PARENTING', 'DF Glue Board', 5.0, 1, 0.0),
 (1, '100+', 'PERSONALIZATION', 'RETRO Shocked DW-6000', 5.0, 13, 0.0),
 (1, '500+', 'PHOTOGRAPHY', 'DC-014', 5.0, 3, 0.0),
 (1, '500+', 'PRODUCTIVITY', 'Ek Vote', 5.0, 43, 0.0),
 (1, '1,000+', 'SHOPPING', 'Dine In CT - Food Delivery', 5.0, 4, 0.0),
 (1, '500+', 'SOCIAL', 'iCard BD Plus', 5.0, 2, 0.0),
 (1, '100+', 'SPORTS', 'FK Dedinje BGD', 5.0, 36, 0.0),
 (1, '5,000+', 'TOOLS', 'CL Keyboard - Myanmar Keyboard (No Ads)', 5.0, 24, 0.0),
 (1, '500+', 'TRAVEL_AND_LOCAL', "CJ's Coffee Cafe", 5.0, 6, 0.0),
 (1, '100+', 'VIDEO_PLAYERS', 'AK Lodi Films', 4.9, 23, 0.0),
 (1, '1,000,000+', 'WEATHER', 'Weather forecast', 4.8, 159455, 0.0)]

<a id="insights"></a>
## Insights

1. Paid apps exhibit higher ratings but lower installs 
2. Categories such as dating, maps and navigation, tools, video players, and lifestyle do not meet the market needs, indicating potential market opportunities
3. A new app should aim for an average rating above 4.19
4. The high volume of apps in communication, social, video player, productivity and game categories indicates a potentially saturated market 

<a href="#table_of_contents">Navigate to contents</a>